src/miner.cpp

```diff
@@ -358,6 +370,12 @@ void BlockAssembler::addPackageTxs()
 
     CTxMemPool::indexed_transaction_set::index<ancestor_score>::type::iterator mi = mempool.mapTx.get<ancestor_score>().begin();
     CTxMemPool::txiter iter;
+
+    // Limit the number of attempts to add transactions to the block when it is
+    // close to full.
+    const int64_t MAX_CONSECUTIVE_FAILURES = 1000;
```

Comment:
Yes, well I also tried setting this to 100 and the performance was not really distinguishable.